### DISCUSSION TITLE
Add support for collection syncing.

### DISCUSSION THREAD

**Discussion by lukeholder** [nmaier95/craft-shopify-product-fetcher#111] [Ideas]
### Description

- Sync Collection just like we sync products.
- Allow the collection to return the related product elements.

https://github.com/craftcms/shopify/issues/110

**Comment by linear** [comment]
<p><a href="https://linear.app/craftcms/issue/PT-1662/add-support-for-collection-syncing">PT-1662 Add support for collection syncing.</a></p>

**Comment by plcdnl** [comment]
We need it, please 🙏

**Comment by andrevvm** [comment]
This is a major deciding factor for us between using Craft or Sanity for Shopify projects. Any plans for the near future? Thank you!

**Reply by JayBox325** [reply]
What did you end up going with? How have you found it with Sanity? I'm in the same boat with choosing the tooling to power my Astro frontends.

**Comment by JayBox325** [comment]
I've been looking into this and what data is available from the Shopify webhooks and all the collection webhook provides is:

```json
{
    "id": XXX,
    "handle": "new-test-collection",
    "title": "New Test collection",
    "updated_at": "2025-02-27T09:47:06-05:00",
    "body_html": "",
    "published_at": "2025-02-27T09:47:06-05:00",
    "sort_order": "best-selling",
    "template_suffix": "",
    "published_scope": "global",
    "admin_graphql_api_id": "gid://shopify/Collection/XXX"
}
```

So... not much. All of the sorting of what products are filtered into a collection happen elsewhere on Shopify. So, realistically, the only reason why you'd want to bring collections across is if you want to add a page-builder Matrix  to your collections routes.

I wanted to migrate collections to Craft because I don't like the single-level routing for Shopify collections and if I could do it in Craft I would be able to make routes such as `/clothing/accessories/hats/beanies` but Shopify makes them separate routes.

However, in order to fully achieve this you'd need to use the Shopify webhooks to migrate almost everything across from Shopify to Craft including tags, options, metaobject and options, which was proving to be problematic with relying on the Webhook to update Craft fast enough to make this workable.

I might come back and try to migrate everything so I can make my collections and handle 100% of the routing in Craft, but for now I'm leaving collection organisation on Shopify.

EDIT: I have decided to import collections, just to make them available for use with Verbb's Navigation plugin to allow authors to add collections to the URL without having to find their URLs.

**Comment by meisbellajack** [comment]
This is a massive addition. As stores scale, 'Manual Option Management' becomes the biggest friction point. Merchants don't want to set up the same Conditional Logic or Image Swatches 50 times across 50 products they want to set it once at the Collection level and have it ripple through automatically.

A few architectural thoughts for this implementation:

Inheritance vs. Overrides: It would be great to see a 'Master Template' approach where a Collection can 'push' option settings to all products within it, but still allow for individual product 'overrides' if a specific item needs a unique file upload or engraving field.

Smart Collections Integration: For 2026, we should ensure this works with Automated Collections (based on tags/metafields). If a product gets tagged with 'Customizable,' the collection sync should automatically trigger the extra variant options on that PDP.

The '3-Option' Constraint: As many of you know, Shopify’s native 3-option limit is usually what breaks these syncs. If a collection has products with 2 options (Size, Color) and others with 3 (Size, Color, Material), syncing a 4th global option (like 'Gift Wrap') natively is impossible.

How we’re solving this at Misk:
We built [Misk Smart Product Options](https://apps.shopify.com/misk-smart-product-options?) specifically as a native App Embed to handle this 'Global Syncing' layer. Our users can apply a 'Smart Logic' rule to an entire Collection in one click, bypassing the 3-option limit while keeping the mobile site speed lightning-fast.

If we can align this GitHub sync support with a 'Global Template' logic, it would save merchants hours of manual work and keep their storefronts looking great and beautiful without technical debt.
